osu.Game/Screens/Edit/Editor.cs

```diff
@@ -300,6 +302,9 @@ private void load(OsuConfigManager config)
                 dependencies.CacheAs<IEditorChangeHandler>(changeHandler);
             }
 
+            newChangeHandler = new NewBeatmapEditorChangeHandler(editorBeatmap);
+            dependencies.CacheAs(newChangeHandler);
```

Comment:
I lined out a bit on how to migrate this in parts in https://github.com/ppy/osu/pull/30255 and that's unchanged for this PR as well.

I can implement something that allows the undo history of the old and new change handlers to be interleaved, so we can have some parts undo with the new system and some parts with the old system.

However we'll have to migrate all the hitobject stuff for all rulesets at once because its not possible to undo a hitobject with the old system and then with the new system. The old system replaces the hitobject with a new object on undo so references break.